File: Selenium/Activities/selenium_Activity7.java
package selenium_session1;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

public class selenium_Activity7 
{
	public static void main(String[] arg)
	{
		//create a WebDriver instance
		WebDriver driver = new FirefoxDriver();
				
		//go to a web page
		driver.get("https://training-support.net/webelements/dynamic-controls");
		
		//do stuff on the page
		System.out.println("Title of the page is : " + driver.getTitle());
		
		WebElement element = driver.findElement(By.id("textInput"));
				
		System.out.println("textInput is enabled : " + element.isEnabled());
		System.out.println("textInput TagName : " + element.getTagName());
		
		driver.findElement(By.xpath("//*[@id=\"textInputButton\"]")).click();
		
		System.out.println("textInput is enabled : " + element.isEnabled());
		
		if (!element.isEnabled())
		{
			driver.findElement(By.xpath("//*[@id=\"textInputButton\"]")).click();
		}
		 // Type something in to it
		element.sendKeys("Example text");
        System.out.println(element.getDomProperty("value"));

		
		//close the browser
		driver.quit();
	}

}

File: Selenium/Activities/selenium_activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By

#create WebDriver instance
with webdriver.Firefox() as driver:

    #go to web page
    driver.get("https://training-support.net/webelements/dynamic-controls")

    #do the stuff on the page
    pageTitle = driver.title
    print("Title of the page is : " + pageTitle)

    #Find the text field.
    textInput = driver.find_element(By.ID, "textInput")

    #Check if the text field is enabled and print it.
    print("textInput is enabled: " , textInput.is_enabled())
    
    #Click the "Enable Input" button to enable the input field.
    driver.find_element(By.XPATH, "//*[@id=\"textInputButton\"]").click()
    
    #Check if the text field is enabled again and print it.
    print("textInput is enabled: " , textInput.is_enabled())


    if textInput.is_enabled() != True:
        driver.find_element(By.XPATH, "//*[@id=\"textInputButton\"]").click()
		
    textInput.send_keys("Example text")
    print(textInput.get_attribute("value"))

    #close the browser
    driver.quit()